src/styles/index.scss

```diff
@@ -1,3 +1,79 @@
+$days-count: 31;
+$day-size: 100px;
+$day-gap: 1px;
+$day-color: #eee;
+$border: 1px solid black;
+$calendar-columns: 7;
+$font-size: 30px;
+$day-font-family: Arial, Helvetica, sans-serif;
+
 body {
   margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: $calendar-columns * $day-size + ($calendar-columns - 1) + $day-gap +
+    $calendar-columns * 2;
+  counter-reset: day 0;
+}
+
+@for $days from 28 through 31 {
+  .calendar--month-length-#{$days} {
+    .calendar__day:nth-child(n + #{$days + 1}) {
+      display: none;
+    }
+  }
+}
+
+.calendar__day {
+  width: $day-size;
+  height: $day-size;
+  background-color: $day-color;
+  border: $border;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  position: relative;
+  box-sizing: border-box;
+  margin-right: $day-gap;
+  margin-bottom: $day-gap;
+
+  &::before {
+    content: counter(day);
+    counter-increment: day;
+    font-size: $font-size;
+    font-family: $day-font-family;
+  }
+
+  &:hover {
+    cursor: pointer;
+    background-color: #ffbfcb;
+    transform: translateY(-20px);
+    transition-duration: 0.5s;
```

Comment:
The `transition-duration` should be specified within the default state of the element, not on the `:hover` state. This ensures that the transition is smooth in both directions. Move `transition-duration: 0.5s;` to the `.calendar__day` block.